.github/workflows/publish-image.yml

```diff
@@ -0,0 +1,66 @@
+# .github/workflows/build-and-push-image.yml
+
+name: Build and Push Docker Image
+
+on:
+  workflow_call:
+    inputs:
+      artifact-url:
+        description: 'The URL of the Artifact WAR file to be downloaded'
+        required: true
+        type: string
+      runner-label:
+        required: false
+        default: 'ubuntu-24.04'
+        type: string
+      maven-repo:
+        required: false
+        default: 'code-artifact'
+        type: string
+
+jobs:
+  publish-image:
+    runs-on: ${{ inputs.runner-label }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
```

Comment:
I think this is considered good practice because tags may be changed, and these commit ids can't. On the other the hand, the uglyness of that always hold me back. Is it actually true that we can't trust the maintainers of something like 'actions/checkout', and that they may put something malicious in @v4? (or @v4.2.2 for that matter)

And of course it may also work the other way around. If the found a vulnarability in there thing, they patch it, and set the new tag @v4.2 to the lastest patch version, our builds will profit without us having to change a thing.

Reply:
@gb-beng ?
I am also agree with @mihxil